sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -221,6 +222,19 @@ public PlannerResult plan() throws SqlParseException, ValidationException, RelCo
     final SqlNode validatedQueryNode = planner.validate(parameterizedQueryNode);
     final RelRoot rootQueryRel = planner.rel(validatedQueryNode);
 
+    if (parsed.getInsertNode() != null) {
```

Comment:
this can be a part of the `insertNode` block in `buildQueryMaker`. We already have a validation method for DS there, possibly this one can be in a `validateInsertSource` like method

Reply:
Moved the code here